src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -153,5 +97,67 @@ private static IsolatedComponentLoadContext GetIsolatedComponentLoadContext(stri
 
             return alc;
         }
+
+        private static IntPtr InternalGetFunctionPointer(AssemblyLoadContext alc,
+                                                         string typeName,
+                                                         string methodName,
+                                                         IntPtr delegateTypeNative)
+        {
+            // Create a resolver callback for types.
+            Func<AssemblyName, Assembly> resolver = name => alc.LoadFromAssemblyName(name);
+
+            // Determine the signature of the type. There are 3 possibilities:
+            //  * No delegate type was supplied - use the default (i.e. ComponentEntryPoint).
+            //  * A sentinel value was supplied - the function is marked UnmanagedCallersOnly. This means
+            //      a function pointer can be returned without creating a delegate.
+            //  * A delegate type was supplied - Load the type and create a delegate for that method.
+            Type? delegateType;
+            if (delegateTypeNative == IntPtr.Zero)
+            {
+                delegateType = typeof(ComponentEntryPoint);
+            }
+            else if (delegateTypeNative == (IntPtr)(-1))
+            {
+                delegateType = null;
+            }
+            else
+            {
+                string delegateTypeName = MarshalToString(delegateTypeNative, nameof(delegateTypeNative));
```

Comment:
```suggestion
                string delegateTypeName = Marshal.PtrToStringAuto(delegateTypeNative)!;
```

Reply:
Ok, you made me look up the source code for `Marshal.PtrToStringAuto` since it isn't documented that it calls `PtrToStringUni` or `PtrToStringUTF8` based on the platform. From there, I see that `Marshal.PtrToStringUni` returns null if `IsNullOrWin32Atom`. So I think I need to pull this back out to its own method, and the `MarshalToString` method should stay as it is.

https://github.com/dotnet/runtime/blob/f7458a82338a60603d807106511af32452d40703/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs#L58-L63

Reply:
I changed my mind. I would rather treat this case like AV and leave the code as is because separating it into another function was too ugly.